webpack/stories/docs/state-management.stories.mdx

```diff
@@ -0,0 +1,73 @@
+import { Meta } from '@theforeman/stories';
+
+<Meta
+  title="Introduction/Where to Put Stuff"
+  parameters={{
+    storyWeight: 132,
+  }}
+/>
+
+# State Management in Foreman & Plugins
+
+Where do I put data so that it's accessible by the React components that need it?
+
+Currently Foreman has a few places for storing data:
+
+* Component state
+* Redux store
+* ForemanContext
+* Global `window` object
+* Other React Context
+* localStorage and sessionStorage
+* Routing
+
+The following sections will outline
+
+*  The architecture around how we use each method
+*  Current opinions on best practices.
+
+## Local component state
+
+This includes anything that uses `useState` or `useReducer` hooks.  Or, in class components, `this.state`.  See the React docs for more info.
+
+This is the best place for any data that is _used only locally by the component_: for example
```

Comment:
What about state that affects the direct children? I think that could also a good use of local state in some cases.

Reply:
oh right, of course!  I'll mention that as well.